### PR TITLE
Implement the root manifest

### PR DESCRIPTION
**Pull request by yagoulas**
## Purpose

This fixes the fact that contrary to windows we don't have a root (implicit) manifest. Windows do that by loading Microsoft.Windows.SystemCompatible manifest for every process. To remedy this we have hacks here and there.

## Proposed changes

Wine's activation context code have some special cases for the process activation context. For the two of these cases I did the same trick and added support for the implicit activation context there 

This needs a lot of testing to make sure that nothing breaks from this. In subsequent commits I'll remove hacks we had for it.

## Future work

After this gets merged I'll create a new PR to make it possible to use a different SystemCompatible based on the compatibility tag of the application manifest. This means that it will be possible to implicitly add a new assembly (aka a collection of dlls) to processes that explicitly target vista+ window versions. Such an assembly could include apisets among others.

### CODE REVIEW

File: dll/win32/gdiplus/x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef.manifest [hunk 1]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
-  <assemblyIdentity type="win32" name="Microsoft.Windows.GdiPlus" version="1.0.100.0" processorArchitecture="x86" />
+  <assemblyIdentity type="win32" name="Microsoft.Windows.GdiPlus" version="1.1.7601.23038" processorArchitecture="" publicKeyToken="6595b64144ccf1df"/>
```
Comment (learn-more) [inline dll/win32/gdiplus/x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef.manifest]: sometimes processorArchitecture is empty, sometimes it's `*`, i'm not sure which is best?

File: sdk/lib/rtl/actctx.c [hunk 2]
```diff
@@ -4665,6 +4667,21 @@ void actctx_init(void)
     {
         process_actctx = check_actctx(handle);
     }
+
+    ctx.dwFlags  = 0;
+    ctx.hModule  = NULL;
+    ctx.lpResourceName = NULL;
+    ctx.lpSource = buffer;
+    wcscpy( buffer, SharedUserData->NtSystemRoot );
+    wcscat( buffer, L"\\winsxs\\manifests\\x86_microsoft.windows.systemcompatible_6595b64144ccf1df_5.82.2600.2982_none_deadbeef.manifest");
```
Comment (learn-more) [inline sdk/lib/rtl/actctx.c]: StringCch, or even `RTL_UNICODE_STRING_BUFFER` that will grow dynamically? (See `SdbpMatchExe` for an example)

File: sdk/lib/rtl/actctx.c [hunk 3]
```diff
@@ -4665,6 +4667,21 @@ void actctx_init(void)
...
+    if (NT_SUCCESS(RtlCreateActivationContext(0, (PVOID)&ctx, 0, NULL, NULL, &handle)))
+    {
+        implicit_actctx = check_actctx(handle);
+    }
+    else
+    {
+        DPRINT1("Failed to create the implicit act ctx!!!\n");
```
Comment (learn-more) [inline sdk/lib/rtl/actctx.c]: Please include NTSTATUS

File: win32ss/user/user32/windows/class.c [hunk 4]
```diff
@@ -37,49 +37,6 @@ struct wndclass_redirect_data
...
-    while (min <= max)
-    {
-        int res, pos = (min + max) / 2;
-        if (!(res = strcmpiW( name, classesW[pos] ))) return classesW[pos];
-        if (res < 0) max = pos - 1;
-        else min = pos + 1;
-    }
-    return NULL;
-}
-
```
Comment (learn-more) [inline win32ss/user/user32/windows/class.c]: Nice!

File: win32ss/user/user32/windows/class.c [hunk 5]
```diff
@@ -160,24 +117,6 @@ ClassNameToVersion(
...
-         */
-        VersionedClass = is_comctl32_class(SectionName.Buffer);
-        if (VersionedClass)
-        {
-            if (pContext) *pContext = 0;
-            if (plpLibFileName) *plpLibFileName = L"comctl32";
-        }
-    }
-#endif
-
```
Comment (learn-more) [inline win32ss/user/user32/windows/class.c]: Nice!

File: dll/win32/gdiplus/x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef.manifest (hunk 1 above)
Comment (yagoulas) [reply dll/win32/gdiplus/x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef.manifest]: I am not really sure. I think we should keep x86 and x64 copies.

File: dll/win32/gdiplus/x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef.manifest [hunk 6]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<?xml version="1.0" encoding="UTF-8" standalone="yes?>
```
Comment (SergeGautherie) [inline dll/win32/gdiplus/x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef.manifest]: Unexpected change?

File: media/CMakeLists.txt [hunk 7]
```diff
@@ -12,3 +12,5 @@ add_cd_file(FILE ${CMAKE_CURRENT_SOURCE_DIR}/drivers/etc/networks DESTINATION re
 add_cd_file(FILE ${CMAKE_CURRENT_SOURCE_DIR}/drivers/etc/protocol DESTINATION reactos/system32/drivers/etc FOR all)
 add_cd_file(FILE ${CMAKE_CURRENT_SOURCE_DIR}/drivers/etc/services DESTINATION reactos/system32/drivers/etc FOR all)
 add_cd_file(FILE ${CMAKE_CURRENT_SOURCE_DIR}/sounds/ReactOS_LogOn.wav DESTINATION reactos/media FOR all)
+
+add_cd_file(FILE ${CMAKE_CURRENT_SOURCE_DIR}/x86_microsoft.windows.systemcompatible_6595b64144ccf1df_5.82.2600.2982_none_deadbeef.manifest DESTINATION reactos/winsxs/manifests FOR all)
```
Comment (SergeGautherie) [inline media/CMakeLists.txt]: Missing EoF newline?

Comment (learn-more) [reply media/CMakeLists.txt]: No?

**Pull request merged**